eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/PageableWrapperProvider.cs

```diff
@@ -124,10 +125,15 @@ private MethodProvider BuildAsPagesMethod()
 
             // Build the method body using foreach and yield return pattern
             var pageType = new CSharpType(typeof(Page<>), _tType);
+            var asPagesInvocation = _sourceField.Invoke("AsPages", [continuationTokenParam, pageSizeHintParam]);
+            var pages = _isAsync
+                // we can't write ConfigureAwait for a non-async invoke, this is a workaround
+                ? asPagesInvocation.Invoke(nameof(TaskAsyncEnumerableExtensions.ConfigureAwait), [False], null, false, false, extensionType: typeof(TaskAsyncEnumerableExtensions))
+                : asPagesInvocation;
```

Comment:
I think we should centralize this kind of complicated logic of expressions into snippets.
For instance, in this case, we could write a `PageableSnippets` and a `AsyncPageableSnippets` to handle such logic gracefully.

And the logic here should look like:
```
var pages = _isAsync ? _sourceField.As<Pageable<object>>().AsPages() : _sourceField.As<AsyncPageable<object>>().AsPages(); 
```
the `AsyncPageableSnippets.AsPages` should contain the `configureawait` invocation

Reply:
As offline discussed, this is the only usage, we can keep it as it is.